lib/Service/RoomService.php

```diff
@@ -362,6 +366,56 @@ public function setAvatar(Room $room, string $avatar): bool {
 		return true;
 	}
 
+	public function startRecording(Room $room, int $status): void {
+		$availableRecordingTypes = [Room::RECORDING_VIDEO, Room::RECORDING_AUDIO];
+		if (!in_array($status, $availableRecordingTypes)) {
+			throw new InvalidArgumentException('status');
+		}
+		if ($room->getCallRecording() !== Room::RECORDING_NONE) {
+			throw new InvalidArgumentException('recording');
+		}
+		if (!$room->getActiveSince() instanceof \DateTimeInterface) {
+			throw new InvalidArgumentException('call');
+		}
+		$this->setCallRecording($room, $status);
+	}
+
+	public function stopRecording(Room $room): void {
+		if ($room->getCallRecording() === Room::RECORDING_NONE) {
+			throw new InvalidArgumentException('recording');
+		}
+		$this->setCallRecording($room);
+	}
+
+	/**
+	 * @param Room $room
+	 * @param integer $status 0 none|1 video|2 audio
+	 */
+	public function setCallRecording(Room $room, int $status = Room::RECORDING_NONE): void {
+		if (!$this->config->isRecordingEnabled()) {
```

Comment:
```suggestion
		if (!$this->config->isRecordingEnabled() && $status !== Room::RECORDING_NONE) {
```

We should always allow to end the recording, even if the admin changed the setting in the meantime.